lib/plugins/audit.js

```diff
@@ -198,67 +201,80 @@ function auditLogger(opts) {
-                        typeof req.query === 'function'
-                            ? req.query()
-                            : req.query,
-                    method: req.method,
-                    url: req.url,
-                    headers: req.headers,
-                    httpVersion: req.httpVersion,
-                    trailers: req.trailers,
-                    version: req.version(),
-                    body: opts.body === true ? req.body : undefined,
-                    timers: timers,
-                    connectionState:
-                        req.connectionState && req.connectionState()
-                };
-            },
-            res: function auditResponseSerializer(res) {
-                if (!res) {
-                    return false;
-                }
+    var reqSerializer = function auditRequestSerializer(req) {
+        if (!req) {
+            return false;
+        }
+
+        var timers = {};
+        (req.timers || []).forEach(function forEach(time) {
+            var t = time.time;
+            var _t = Math.floor(1000000 * t[0] + t[1] / 1000);
+            timers[time.name] = (timers[time.name] || 0) + _t;
+        });
+        return {
+            // account for native and queryParser plugin usage
+            query: typeof req.query === 'function' ? req.query() : req.query,
+            method: req.method,
+            url: req.url,
+            headers: req.headers,
+            httpVersion: req.httpVersion,
+            trailers: req.trailers,
+            version: req.version(),
+            body: opts.body === true ? req.body : undefined,
+            timers: timers,
+            connectionState: req.connectionState && req.connectionState()
+        };
+    };
 
-                var body;
+    if (opts.serializers && opts.serializers.req) {
+        reqSerializer = opts.serializers.req;
+    }
+
+    var resSerialzer = function auditResponseSerializer(res) {
+        if (!res) {
+            return false;
+        }
 
-                if (opts.body === true) {
-                    if (res._body instanceof HttpError) {
-                        body = res._body.body;
-                    } else {
-                        body = res._body;
-                    }
-                }
+        var body;
 
-                return {
-                    statusCode: res.statusCode,
-                    headers: getResponseHeaders(res),
-                    trailer: res._trailer || false,
-                    body: body
-                };
+        if (opts.body === true) {
+            if (res._body instanceof HttpError) {
+                body = res._body.body;
+            } else {
+                body = res._body;
             }
         }
+
+        return {
+            statusCode: res.statusCode,
+            headers: getResponseHeaders(res),
+            trailer: res._trailer || false,
+            body: body
+        };
+    };
+
+    if (opts.serializers && opts.serializers.res) {
```

Comment:
Style nit: might we be able to collapse these three if checks into a single assignment? Something like:

```js
var serializers = Object.assign({}, DEFAULT_SERIALIZERS, opts.serializers);
```

Reply:
This is a good idea.

Reply:
Made the change, what do you think now @DonutEspresso?